### DISCUSSION TITLE
MS SQL Server Connection using Python

### DISCUSSION THREAD

**Discussion by ghost** [AzureAD/microsoft-authentication-library-for-python#442] [Q&A]
**Describe the bug**
There is no sample code for connecting to ms sql server using python . Please provide some sample for same.

**Comment by rayluo** [comment]
* There was a brief recipe of [how to use ADAL Python to acquire token and then connect to Azure SQL database](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/Connect-to-Azure-SQL-Database). (Note that, ADAL Python is deprecated at this point, but the overall concept of obtaining a token should still be applicable, so, you could swap the "ADAL client credential sample" mentioned in that recipe, with the [MSAL client credential sample](https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/1.16.0/sample/confidential_client_secret_sample.py).)
* A possibly [relevant Q&A in MSAL .Net](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/894)

Hope that helps.
